app/ts/background/windows/personalSign.ts

```diff
@@ -171,6 +171,27 @@ export async function craftPersonalSignPopupMessage(ethereumClientService: Ether
 			}
 		} as const
 	}
+	if (namedParams.param.primaryType === 'SafeTx') {
+		const parsed = SafeTx.parse(namedParams.param)
+		return {
+			method: 'popup_personal_sign_request',
+			data: {
+				originalParams,
+				...basicParams,
+				type: 'SafeTx' as const,
+				message: parsed,
+				account,
+				addressBookEntries: {
+					to: getAddressMetaData(parsed.message.to, userAddressBook),
+					gasToken: await getTokenMetadata(ethereumClientService, parsed.message.gasToken),
+					refundReceiver: getAddressMetaData(parsed.message.refundReceiver, userAddressBook),
+					verifyingContract: getAddressMetaData(parsed.domain.verifyingContract, userAddressBook),
+				},
+				quarantine: false,
+				quarantineCodes: [],
+			}
+		} as const
```

Comment:
casting as `const` may be omitted as the function return type is already declared